gallery/gallery/lib/demos/material/menu_demo.dart

```diff
@@ -0,0 +1,303 @@
+}
+
+class _MenuDemoState extends State<MenuDemo> {
+  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  SimpleValue _simpleValue;
+  List<CheckedValue> _checkedValues;
+
+  @override
+  void initState() {
+    super.initState();
+    _simpleValue = SimpleValue.two;
+    _checkedValues = [CheckedValue.three];
+  }
+
+  void showInSnackBar(String value) {
+    _scaffoldKey.currentState.showSnackBar(SnackBar(
+      content: Text(value),
+    ));
+  }
+
+  void showAndSetMenuSelection(BuildContext context, SimpleValue value) {
+    setState(() {
+      _simpleValue = value;
+    });
+    showInSnackBar(GalleryLocalizations.of(context)
+        .demoMenuSelected(simpleValueToString(context, value)));
+  }
+
+  void showCheckedMenuSelections(BuildContext context, CheckedValue value) {
+    if (_checkedValues.contains(value)) {
+      setState(() {
+        _checkedValues.remove(value);
+      });
+    } else {
+      setState(() {
+        _checkedValues.add(value);
+      });
+    }
+
+    showInSnackBar(
+      GalleryLocalizations.of(context).demoMenuChecked(
+        _checkedValues.map((value) => checkedValueToString(context, value)),
+      ),
+    );
+  }
+
+  String simpleValueToString(BuildContext context, SimpleValue value) => {
+        SimpleValue.one: GalleryLocalizations.of(context).demoMenuItemValueOne,
+        SimpleValue.two: GalleryLocalizations.of(context).demoMenuItemValueTwo,
+        SimpleValue.three:
+            GalleryLocalizations.of(context).demoMenuItemValueThree,
+      }[value];
+
+  String checkedValueToString(BuildContext context, CheckedValue value) => {
+        CheckedValue.one: GalleryLocalizations.of(context).demoMenuOne,
+        CheckedValue.two: GalleryLocalizations.of(context).demoMenuTwo,
+        CheckedValue.three: GalleryLocalizations.of(context).demoMenuThree,
+        CheckedValue.four: GalleryLocalizations.of(context).demoMenuFour,
+      }[value];
+
+  bool isChecked(CheckedValue value) => _checkedValues.contains(value);
+
+  @override
+  Widget build(BuildContext context) {
+    Widget menu;
+    switch (widget.type) {
+      case MenuDemoType.contextMenu:
+        // BEGIN menuDemoContext
+
+        // Pressing the PopupMenuButton on the right of this item shows
+        // a simple menu with one disabled item. Typically the contents
+        // of this "contextual menu" would reflect the app's state.
+        menu = ListTile(
+          title: Text(GalleryLocalizations.of(context)
+              .demoMenuAnItemWithAContextMenuButton),
+          trailing: PopupMenuButton<String>(
+            padding: EdgeInsets.zero,
+            onSelected: (value) => showInSnackBar(
+                GalleryLocalizations.of(context).demoMenuSelected(value)),
+            itemBuilder: (context) => <PopupMenuItem<String>>[
+              PopupMenuItem<String>(
+                value: GalleryLocalizations.of(context).demoMenuItemValueOne,
```

Comment:
This should be `demoMenuContextMenuItemOne`

Reply:
It should actually be `demoMenuItemValueOne`.  The value is different from the label in the original demo, and it is then shown in the snackbar.

Reply:
The value and the label is different here so it is actually right.

Reply:
But I guess it doesn't matter really so I can change it